### DISCUSSION TITLE
Decouple gui from cli

### DISCUSSION THREAD

**Discussion by webysther** [TPC-Council/HammerDB#109] [Q&A] [answered]
Decoupling gui from cli in two projects (and repositories) sounds better to focus in diferrents use cases.

Another field to consider is using [electron](https://www.electronjs.org/) as base to a new gui, focusing all features using a API through CLI with a possible global flag `--output=json`.

**Comment by sm-shaw** [comment]
Having 2 projects one for GUI and one for CLI could open up the potential for the performance to be different between the 2.  It is also very important to have a GUI that can be installed and run on both Linux and Windows without a lot of dependencies.  At present underneath the GUI and the CLI it is exactly the same code - so when you are running tests you know they will be the same. (You also develop a feature just once and it is available in both). 

Nevertheless using hammerdbws would be a good initial starting point if someone wanted to look at developing a new GUI over this as that provides an HTTP interface to the already highly scalable engine.  A key issue right now is 4KHD support as detailed in this issue GUI HD display #19. It is worth noting that one of the main things that the GUI adds is the settings to select the right options when configuring a workload, the CLI assumes more advanced knowledge.

**Comment by sm-shaw** [comment, marked as answer]
v4.0 has been released and includes the GUI HD display #19. There are no plans to decouple the GUI and CLI. However the WS version provides the opprotunity to interact using json for anyone who wishes to do so.
